Cleaning of internal cachés Orika 's

ettifouri opened the pull request:
Hi dear Orika team,

   In a java project, We use Orika  to duplicate a large volume of a object A to another B. 

  When the operation is complete, we see there are always an important cosomation memory.

 Apparently, Orika uses some internal caches to save the duplicated bean; So, is there any way to clean this caches manually ?  

Note: We are using the 1.3.5 Orika version.

Thank you for the help
## Cordially

ETTIFOURI El Hassane

matt-deboer committed:
Added a link to new documentation

matt-deboer committed:
Added set to list case

matt-deboer committed:
Applied test case for introspector bug in jdk 1.6.0_u05, and applied
modified fix.
Added fix for issue #74 which uses the ConcurrentLinkedHashMap from
googlecode as a replacement for CacheLRUHashMap which was not
thread-safe.

elaatifi committed:
Merge branch 'master' of http://github.com/orika-mapper/orika

elaatifi committed:
Update README.md

elaatifi committed:
Update README.md

matt-deboer committed:
Added test case for concurrency issue and fixed bugs related to
concurrent modification in SortedCollection and SortedSet, and with
improper synchronization in DefaultMapperFactory

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

matt-deboer committed:
Added additional concurrency test for unique generation of class names;
added static AtomicInteger in SourceCodeContext to guarantee unique
generated names across a class-loader

matt-deboer committed:
simplified synchronization on DefaultMapperFactory.registerClassMap;
updated uniqueness algorithm on SourceCodeContext to incorporate
nanoTime along with value of AtomicInteger, which should work better in
cases where class/source files are written out

matt-deboer committed:
Added java.sql.Date and java.sql.Time support to standard built-in
converters.

matt-deboer committed:
Added support for java.sql.Timestamp to builtin converters

matt-deboer committed:
Added test case for issue #77, along with fix to DefaultMapperFactory:
broke a-to-b registry up into dynamic and explicit registry; then made
resolution of concrete destination type check for explicit first, then
exact type if concrete, then dynamic

matt-deboer committed:
Cleaned up order of code in lookupConcreteDestinationType to reflect
actual order of search

Commit:
modified algorithm to generate object name suffix

elaatifi committed:
Merge pull request #1 from marcoslacoste/master

Modified algorithm to generate object name suffix on ObjectFactoryGenerator.java

elaatifi committed:
Update default scm url to orikamapper

elaatifi committed:
[maven-release-plugin] prepare release orika-parent-1.4.1

elaatifi committed:
[maven-release-plugin] prepare for next development iteration

elaatifi committed:
Update README.md

elaatifi committed:
Fix a bug related to two way mapping with a long path property

matt-deboer committed:
Applied test case for issue with specified types, and fix (this test
only fails on certain JDKs, specifically hotspotv6 on mac, and v6IBM
on windows, etc.)
Updated some of the test cases that were failing on IBM JDK6:
+CloneableConverter was failing due to IllegalAccessException
+MultiThreadedTestCase was failing upon intentional OOME; added custom
version of this method for IBM jdk
Updated javadocs on Property and NestedProperty

papegaaij committed:
Testcase showing problems with subtype mapping and unenhancing

papegaaij committed:
possible fix for subtype mapping and unenhancing: do not check for existing mappers

matt-deboer committed:
Merge pull request #3 from topicusonderwijs/master

Subtypes not mapped correctly for hibernate proxies

elaatifi committed:
Issue91 Add the possibility to specify elementType by ClassMapBuilder API

elaatifi committed:
Merge branch 'Issue91'

papegaaij committed:
Do not use proxy classes as cache keys

papegaaij committed:
testcase showing problems with cached strategies and proxy classes

elaatifi committed:
Merge pull request #5 from topicusonderwijs/cache-bug

Strategies are cached under proxy classes

ikozar committed:
Apply the pattern chain of responsibilities for choice suitable ClassMapBuilderFactory for the given types

elaatifi committed:
Merge pull request #6 from ikozar/core-ref

Apply the pattern chain of responsibilities for choice suitable ClassMap...

matt-deboer committed:
Added fix for Issue#94;
Also added correction for ClassLoaderLeakageTest -- newer sun jdks will
stop before actually allocating enough memory to clear soft&weak
references, so we have to allocate memory in smaller chunks

matt-deboer committed:
Added fix for issue #95 with test case

matt-deboer committed:
Applied Orika formatter (default formatter was accedentally applied to
last commits)

matt-deboer committed:
Applied alternative approach to resolve and cache mapping strategies
which performs better than recent solution.
Fixed ObjectFactoryGenerator to instantiate using the default
constructor when no source type mapping is matched.
Added AbstractMappingStrategy to factor out base capabilities, including
common toString for debugging.

matt-deboer committed:
Added caliper results by version for before and after recent changes for
1.4.2

matt-deboer committed:
Added fix for issue#98 with test case.
Added fix for issue#97 which uses ConcurrentSkipListMap as the base
collection backing SortedCollection and SortedMap.
Included jsr166x for ability to use the previously mentioned classes in
a 1.5jdk

matt-deboer committed:
Removed jsr166x dependency, since it was compiled for jsk 1.6; included
ConcurrentSkipListMap (and necessary dependent classes from jsr166x)
instead into orika core module.
Included fix for issue#82 with test case.

matt-deboer committed:
Included fix for issue#88; use reflection to detect int or long in
method signature as it has changed in an incompatible way from version
1.2 to 1.3 of this class.

matt-deboer committed:
Added issue#100 enhancement, case-insensitive classmap builder support

elaatifi committed:
[maven-release-plugin] prepare release orika-parent-1.4.2

elaatifi committed:
[maven-release-plugin] prepare for next development iteration

matt-deboer committed:
Cleaned up Convert code -- simplified the default case when nulls are
mapped and we're copying by reference.
Also fixed case where strings were being concatenated with empty string
on assignment (causing performance penalty)

matt-deboer committed:
Removed jdk6 String.isEmpty reference.

elaatifi committed:
Update README.md

matt-deboer committed:
Fix for issue #107 in VariableRef.
Moved jsr166x types into an 'ma.glasnost.orika' sub-package.
Deleted duplicate 'java-hidden' types from the 'core' module.
Updated OrikaSystemProperties and DefaultMapperFactory.Builder to pull
default values for all configurations (including boolean values) from
system properties when present.
Added OpenIntObjectHashMap from cern.colt package, along with usage of a
unique integer index to MappingContext to gain speed improvement in
cacheMappedObject and getMappedObject; also updated MappingContext to
only stored mapped object's by type (since storage by raw class was
never used).
Added NonCyclicMappingContext and got rid of NonCyclicBoundMapperFacade
since it was (incorrectly) using the same singleton context instance for
all mapping requests.

matt-deboer committed:
Added test case for issue #92 and fix.

matt-deboer committed:
Added test case and fix for issue #92

matt-deboer committed:
Fix for DynamicSuite which was not running EclipseJdt tests

matt-deboer committed:
Added initial implementation of source-registered object factories.
Added test case for issue #90 which is fixed by source-specific object
factories.

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

matt-deboer committed:
Applied moved jsr166x types to SortedCollection. Removed unused method
'threadsafeCopy'.

matt-deboer committed:
Added test case for issue #102 and fix to mapper facade by allowing copy
by reference when dest is assignable from immutable source (not only
when equal).
Also fixed PropertyResolverTestCase method (which used to fail, but now
maps successfully).

papegaaij committed:
fix NPE when mapping Map<String, List<MyType>>

elaatifi committed:
Merge pull request #7 from topicusonderwijs/master

NPE when mapping Map<String, List<MyType>>

papegaaij committed:
use superclasses on objectfactory lookup

papegaaij committed:
Testcase for resolving object factories using supertypes and also include java.lang.Object

matt-deboer committed:
Merge pull request #8 from topicusonderwijs/master

lookupObjectFactory should consider superclasses for source type

matt-deboer committed:
Fix MappingContext and Type unique index

matt-deboer committed:
Fix DefaultBoundMapperFacade to use new object factory lookup signature.

Commit:
correctly implement compareTo for type hierarchies

papegaaij committed:
testcase asserting compareTo contract

matt-deboer committed:
Merge pull request #10 from topicusonderwijs/master

Correctly implement compareTo contract for Type

matt-deboer committed:
Changed Comparators for Ordering, and moved the class into the util
package with SortedSet and SortedCollection.
Changed Comparable/Comparator implementations for 'Ordering'
implementation which does not have the same contract as Comparable.
Added SortedCollectionConcurrencyTest, to track down possible issue with
invalid insertion order under concurrency; synchronized the 'add'
methods on SortedCollection and SortedSet to fix bugs shown by this
test.

matt-deboer committed:
Cleaned up imports

elaatifi committed:
Clean up core-new (should not be pushed)

papegaaij committed:
rewrite SortedCollection to use topological sorting

papegaaij committed:
fix converter resolution by making topological sort stable

matt-deboer committed:
Merge pull request #11 from topicusonderwijs/master

Rewrite of SortedCollection to use a Topological sorting

matt-deboer committed:
Added HashMapUtility as a common wrapper for correctly getting
ConcurrentLinkedHashMap

matt-deboer committed:
Attached issue #109 test case and fix.

matt-deboer committed:
Updated debug logging and javadocs for mapping specifications

matt-deboer committed:
Updated to account for change to MappingStrategy causing it to implement
MappedTypePair

matt-deboer committed:
Updated javadocs for MapperFacade

matt-deboer committed:
Updated javadoc

matt-deboer committed:
Fixed accidental name switch between CaseInsensitiveClassMapBuilderTest
and ScoringClassMapBuilderTest.
Added capability to CaseInsensitiveClassMapBuilder to treat names in a
case-insensitive fashion in other field map methods (in addition to
byDefault())

matt-deboer committed:
Uploaded fix for issue 80; find List or Map ancestor when applying
ClassMapBuilderForList/Map and looking up custom property type.

matt-deboer committed:
Updated multi-threaded test to account for byte array overhead (which
was occasionally causing failures)

matt-deboer committed:
Added fix for issue #85, and modified exceptions to avoid doubling
details.

matt-deboer committed:
Added enhancement to ClassMapBuilder to allow passing a MappingDirection
to the byDefault() method; extended this to the other ClassMapBuilder
sub-classes, added test cases.
Changed wording of 'choice' to 'choose' and 'applied' to 'appliesTo' for
better readability.
Updated more javadocs

matt-deboer committed:
Added initial test case for issue#87

Commit:
Applied issue87 test case; applied fix to remove check for abstract type
at generatin time -- since there could be a concrete type registered for
that type (as demonstrated in the test case)

matt-deboer committed:
Added issue #96 test case

matt-deboer committed:
Added test case for issue#44 which has already been resolved.

matt-deboer committed:
Applied enhancement for issue#44 that allows element-level converter to
be specified on a collection or array to collection or array

elaatifi committed:
Add Apache license 2 file

elaatifi committed:
[maven-release-plugin] prepare release orika-parent-1.4.3

elaatifi committed:
[maven-release-plugin] prepare for next development iteration

elaatifi committed:
Update README.md

matt-deboer committed:
Added initial implementation of Filters feature

matt-deboer committed:
Added licenses and corrected javadocs.

elaatifi committed:
Fix package of FilterTestCase

matt-deboer committed:
Fix for issue#112, along with test case.

matt-deboer committed:
Merge branch 'master' into filters

matt-deboer committed:
Added MappingContextFactory to UtilityResolver and to
DefaultMapperFactory.Builder to allow users to customize the
MappingContextFactory used by the DefaultMapperFactory

elaatifi committed:
Fix MapperKey equals method

matt-deboer committed:
Changed MappingContext constructor to public for better usability.

elaatifi committed:
Fix Array or collection to array mapping specification. 

When used for constructor mapping, a local variable get a duplicated name.

elaatifi committed:
Issue 115, fix NPE in MultiOccurrence To MultiOccurrence mapping.

elaatifi committed:
Fix the use of a depricated junit assert

elaatifi committed:
Fix #113  Cannot find a mapper when a child bean infers a generic type of its parent bean

elaatifi committed:
Fix issue #116, Remove SLF4J from orika-deps-included jar and add/relocate concurrentlinkedhashmap

ikozar committed:
DateToStringConverter add Locale

elaatifi committed:
Merge pull request #14 from ikozar/master

DateToStringConverter add Locale

elaatifi committed:
Merge pull request #12 from orika-mapper/filters

Filters: new feature, initial implementation

elaatifi committed:
Add .iml to gitignore

elaatifi committed:
Update copyright headers

elaatifi committed:
Cleanup deprectaed API from tests

elaatifi committed:
Fix javadoc typo

elaatifi committed:
Clean up deprecated API

matt-deboer committed:
Added fix for PersonMappingTestCase -- issue with properly detecting
nested property when a element type is used

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

matt-deboer committed:
Updated ObjectFactoryGenerator to check for mismatch of parameters to
constructor args and throw a descriptive exception telling user to
register an object factory.
Updated JaxbElementTestCase to include a case similar to that of
discussion group; added jaxb-impl dependency to tests.

matt-deboer committed:
Added fix to constructor resolution to avoid resolving constructors for
which we can't map all the arguments with known source properties.

roadrunner2 committed:
Allow subclasses to determine the A- and B-types.

The default code assumes direct subclasses that extend this class with
explicit types. This fails, however, if the subclasses have generic types
which are used in the extends clause. Therefore the type determination
code has been factored out into a separate method which can be overridden
by subclasses if desired.

elaatifi committed:
Update README.md

Commit:
Added test case for issue#118

elaatifi committed:
Merge pull request #16 from SergeyVasilyev/master

Added test case for issue#118

roadrunner2 committed:
Improve the information passed to Filter.shouldMap().

This does this in two ways:
1. provide the source object being filtered too
2. provide the source and destination property paths rather than just
   the final name

elaatifi committed:
Issue 118 Fix failing tests; elements of a collections should be loaded lazily or eagerly and not mix

elaatifi committed:
Fix Issue 118, Keep using unenhance strategy

roadrunner2 committed:
Fix NPE when destination in a comparison is null.

roadrunner2 committed:
Provide more information when descending into nested objects.

MappingContext.beginMapping is now invoked for both customer-mappers as
well as custom-converters. Additionally, it is invoked with several
arguments that provide information about the mapping being started.

This implementation does not use the new information passed in; however,
it is now available to subclasses that need to do more tracking of the
current state.

roadrunner2 committed:
Add missing use of Filter.appliesTo().

Also fixed CustomFilter's implementation of appliesTo() to account for
autoboxing (i.e. handle primitive fields).

roadrunner2 committed:
Factored out some common code.

elaatifi committed:
Merge pull request #20 from roadrunner2/custom-filter-subclassing

Allow subclasses to determine the A- and B-types.

elaatifi committed:
Merge pull request #23 from roadrunner2/context-beginmapping-enhancements

Provide more information when descending into nested objects.

roadrunner2 committed:
Apply filters in multi-occurrence-to-multi-occurrence setting too.

No filtering was being done at the root and intermediate nodes, only on
leaf nodes.

roadrunner2 committed:
Note nested objects in multi-occurrence-to-multi-occurrence setting too.

This completes the work in dd61a4828c so that MappingContext.beginMapping
and endMapping are invoked when descending root and intermediate nodes
too.

roadrunner2 committed:
Provide a NullFilter to simplify the implementation of simple filters.

elaatifi committed:
Merge pull request #19 from roadrunner2/npe-fix

Fix NPE when destination in a comparison is null.

elaatifi committed:
Merge pull request #21 from roadrunner2/filter-applies-to

Add missing use of Filter.appliesTo().

elaatifi committed:
Merge pull request #22 from roadrunner2/filter-shouldmap-enhancement

Filter shouldmap enhancement

elaatifi committed:
Merge pull request #24 from roadrunner2/multi-occurence-improvements

Multi occurence improvements

roadrunner2 committed:
Tighten type bounds on the parameters to filter methods.

The properties being filtered are all subclasses of the filter's A and B
type parameters. This change makes that explicit.

roadrunner2 committed:
Make Filter.filterSource work.

elaatifi committed:
Merge pull request #25 from roadrunner2/filter-source-fixes

Filter source fixes

elaatifi committed:
Fix Issue 130 Inline Property Syntax does not support primitives

matt-deboer committed:
Fix to specification debug printing; restored field tag prefix was lost
in
a
previous change.

matt-deboer committed:
Code cleanup.

matt-deboer committed:
Added one-to-many and many-to-one specs.

matt-deboer committed:
Added handling for special one-to-many and many-to-one specifications.

matt-deboer committed:
Added existsProperty method to allow checking for existence of a
property without triggering exceptions.

matt-deboer committed:
Removed @Overrides from interface methods (since orika still targets
jdk 1.5)

matt-deboer committed:
Applied fixes to specifications for comparison generation.
Added test cases for many-to-one and one-to-many mappings.

matt-deboer committed:
Merge remote branch 'origin/master' into one-to-many

matt-deboer committed:
Merge remote branch 'origin/master' into one-to-many

matt-deboer committed:
Merge pull request #26 from orika-mapper/one-to-many

Added one-to-many, many-to-one mapping capability; fixed related bugs in multi-occurrence mapping

matt-deboer committed:
Removed @Override markers for interface implementations (since Orika
still
targets
jdk
1.5)

matt-deboer committed:
Added fix for issue#119

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

matt-deboer committed:
Added test cases for issue # 121;
Added fixes to: Type: created new method isSelfOrAncestorParameterized,
used by updated PropertyResolver and TypeUtil to properly resolve
parameterized
types
from ancestor(s).
Added fixes to MultiOccurrence generation to handle edge cases where
possible NPEs can occur.

matt-deboer committed:
Added test case and fix for issue 114

matt-deboer committed:
Added test case for issue#126, along with fix to avoid throwing error in
'exclude' method if field does not exist in both types.

matt-deboer committed:
Added fix for issue #128 with test case

matt-deboer committed:
Fix to debug logging for excluded fields.

matt-deboer committed:
Added test case for issue#132.
Added ConstructorParameter, and ConstructorParameterResolver to help in
resolving parameters from constructors of the destination type.

matt-deboer committed:
Added test case and fix for mapping non-assignable properties (in-place)
if possible.

matt-deboer committed:
moved concurrent threads to 25 to avoid frequent failures in TravisCI

papegaaij committed:
Massive perforance improvements for very large mappings

elaatifi committed:
Merge pull request #27 from topicusonderwijs/master

Massive perforance improvements for very large mappings

Commit:
Added optional deferred initialisation to allow subclass constructor arguments to be captured first.

matt-deboer committed:
Merge pull request #28 from srjturner/master

Added optional deferred initialisation to ConfigurableMapper

matt-deboer committed:
updated collectUsedMappers in default mapper factory to use a set
instead of list, and assure removal of th existing mapper from it's own
usedMappers
added synchronization to DefaultConverterFactory to deal with parallel
loading issue#133

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

elaatifi commented:
Can you please move your pull request, we moved Orika from my personal repository to https://github.com/orika-mapper/orika ?

matt-deboer committed:
Added fix for issue#134; using eclipse compiler with classes in the
default package.

matt-deboer committed:
Cleaned up code warnings; removed unused fields

matt-deboer committed:
cleaned up imports, removed unused field.

matt-deboer committed:
Separated MultiLayeredClassloader into it's own suite, as it was being
negatively affected by previously loaded classes; this may indicate a
problem with a statically held reference somewhere.

leonard84 committed:
Added Testcase for Issue 135

elaatifi committed:
Merge pull request #29 from leonard84/issue135

Added Testcase for Issue 135

elaatifi committed:
Fix Issue 135: NPE while mapping to a nested property

matt-deboer committed:
Migrated eclipse tools to use recent versions of eclipse jdt
dependencies.

matt-deboer committed:
Fixed spelling error.

matt-deboer committed:
Removed separate alternate compiler strategy.

matt-deboer committed:
Merge branch 'master' of https://github.com/orika-mapper/orika.git

matt-deboer committed:
Added explicit pre-loading of hidden types.

matt-deboer committed:
Added a copy of tccl instead of using the existing tccl to isolate from
other tests.

matt-deboer committed:
Added javadocs

ettifouri commented:
Hello,

  I have tried moving the pull request to the repository https://github.com/orika-mapper/orika; but, apparently, this is not possible. Whenever Github me references to your personal repository.

   To get help, I've put the question in the Orika wiki : https://code.google.com/p/orika/issues/detail?id=136

Thanks for help.

## Cordially

ETTIFOURI El Hassane

matt-deboer committed:
Removed @Override annotations (since orika still targets jdk 1.5)
